### PR TITLE
Gal 352/invitation planning parameters backend

### PR DESCRIPTION
**Pull request by msandhuNHS**
## Description
https://nhsd-jira.digital.nhs.uk/browse/GAL-209
https://nhsd-jira.digital.nhs.uk/browse/GAL-352

Updated terraform script to match latest currently in develop-7.
Added terraform code for GET, PUT to invitation parameters dynamo table.

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: terraform/environment/dev/resources/lambda/invitationParameters/lambdaHandler/invitationParametersLambda.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  const CONFIG_ID_VALUE = 1;
+
+  const params = {
+    Key: {
+      CONFIG_ID: {
+        N: `${CONFIG_ID_VALUE}`,
+      },
+    },
+    TableName: "InvitationParameters",
+  };
```
Comment (LaithM96) [inline terraform/environment/dev/resources/lambda/invitationParameters/lambdaHandler/invitationParametersLambda.js:20]: more comment would be appreciated for others to understand the code more.

File: terraform/environment/dev/resources/lambda/invitationParametersPostForecastUptake/lambdaHandler/invitationParametersPostForecastUptakeLambda.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  const CONFIG_ID = 1;
+  const forecastUptake = JSON.parse(event.body).forecastUptake;
+
+  const params = {
+    "ExpressionAttributeNames": {
+      "#FORECAST_UPTAKE_VALUE": "FORECAST_UPTAKE"
+    },
+    "ExpressionAttributeValues": {
+      ":forecast_uptake_new": {
+        "N": `${forecastUptake}`
```
Comment (LaithM96) [inline terraform/environment/dev/resources/lambda/invitationParametersPostForecastUptake/lambdaHandler/invitationParametersPostForecastUptakeLambda.js:19]: more comment would be appreciated for others to understand the code more.

File: terraform/environment/dev/resources/lambda/invitationParametersPostQuintiles/lambdaHandler/invitationParametersPostQuintilesLambda.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    "TableName": "InvitationParameters",
+    "UpdateExpression": "SET #Q1 = :q1_new, #Q2 = :q2_new, #Q3 = :q3_new, #Q4 = :q4_new, #Q5 = :q5_new"
+  };
+
+  const command = new UpdateItemCommand(params);
+  const response = await client.send(command);
+
+  if (response.$metadata.httpStatusCode == 200) {
+    responseObject.statusCode = 200;
+    (responseObject.headers = {
```
Comment (LaithM96) [inline terraform/environment/dev/resources/lambda/invitationParametersPostQuintiles/lambdaHandler/invitationParametersPostQuintilesLambda.js:56]: more comment would be appreciated for others to understand the code more.

**Review by LaithM96 — APPROVED** [review]
All checks passed, code looks good and error free, all the ac's met the criteria from the code and the files are correct, happy to approve it, however I have added a suggestion "more comment would be appreciated for others to understand the code more" well done!

File: terraform/environment/dev/resources/lambda/invitationParameters/lambdaHandler/invitationParametersLambda.js (hunk 1 above)
Comment (zainm97) [reply terraform/environment/dev/resources/lambda/invitationParameters/lambdaHandler/invitationParametersLambda.js:20]: The code was self explanatory - there are no complex functions hence why no comments

File: terraform/environment/dev/resources/lambda/invitationParametersPostForecastUptake/lambdaHandler/invitationParametersPostForecastUptakeLambda.js (hunk 2 above)
Comment (zainm97) [reply terraform/environment/dev/resources/lambda/invitationParametersPostForecastUptake/lambdaHandler/invitationParametersPostForecastUptakeLambda.js]: The code was self explanatory - there are no complex functions hence why no comments. It is well documented when using the API. See here for more details https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/

File: terraform/environment/dev/resources/lambda/invitationParametersPostQuintiles/lambdaHandler/invitationParametersPostQuintilesLambda.js (hunk 3 above)
Comment (zainm97) [reply terraform/environment/dev/resources/lambda/invitationParametersPostQuintiles/lambdaHandler/invitationParametersPostQuintilesLambda.js]: https://docs.aws.amazon.com/AmazonS3/latest/userguide/cors.html

**Review by LaithM96 — APPROVED** [review]
happy with comments added, approved.

**Comment by Abdul-NHS** [comment]
LGTM, there are some values that should be parameterised for example in lambdaHandler/invitationParametersPutForecastUptakeLambda.js. Happy to approve.

**Review by Abdul-NHS — APPROVED** [review]

**Pull request merged**